src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+                    // keep track of total slashed magnitude
+                    slashedMagnitude += slashedAmount;
+                }
+            }
+
+            // 3. update totalMagnitude, get total magnitude and subtract slashedMagnitude
+            // this will be reflected in the conversion of delegatedShares to shares in the DM
+            Snapshots.History storage totalMagnitudes = _totalMagnitudeUpdate[operator][strategies[i]];
+            totalMagnitudes.push({key: uint32(block.timestamp), value: totalMagnitudes.latest() - slashedMagnitude});
+        }
+    }
+
+    /**
+     * @notice Called by an operator to cancel a salt that has been used to register with an AVS.
+     *
+     * @param salt A unique and single use value associated with the approver signature.
+     */
+    function cancelSalt(
+        bytes32 salt
+    ) external override {
+        // Mutate `operatorSaltIsSpent` to `true` to prevent future spending.
+        operatorSaltIsSpent[msg.sender][salt] = true;
+    }
+
+    /**
+     * @notice Helper for setting an operators allocation delay.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     */
+    function _setAllocationDelay(address operator, uint32 delay) internal {
+        require(delay != 0, InvalidDelay());
+
+        AllocationDelayInfo storage delayInfo = _allocationDelayInfo[operator];
+
+        bool pendingInEffect = delayInfo.pendingDelay != 0 && block.timestamp >= delayInfo.pendingDelayEffectTimestamp;
+
+        if (pendingInEffect) {
+            delayInfo.delay = delayInfo.pendingDelay;
+        }
+
+        delayInfo.pendingDelay = delay;
+        delayInfo.pendingDelayEffectTimestamp = uint32(block.timestamp + ALLOCATION_DELAY_CONFIGURATION_DELAY);
+
+        emit AllocationDelaySet(operator, delay);
+    }
+
+    /**
+     * @notice For a single strategy, update freeMagnitude by adding completable pending free magnitudes
+     * @param operator address to update freeMagnitude for
+     * @param strategy the strategy to update freeMagnitude for
+     * @param numToComplete the number of pending free magnitudes deallocations to complete
+     * @dev read through pending free magnitudes and add to freeMagnitude if completableTimestamp is >= block timestamp
+     * In addition to updating freeMagnitude, updates next starting index to read from for pending free magnitudes after completing
+     */
+    function _updateFreeMagnitude(address operator, IStrategy strategy, uint16 numToComplete) internal {
+        OperatorMagnitudeInfo storage info = operatorMagnitudeInfo[operator][strategy];
+        (uint64 freeMagnitudeToAdd, uint192 completed) =
+            _getPendingFreeMagnitude(operator, strategy, numToComplete, info.nextPendingFreeMagnitudeIndex);
+        info.freeMagnitude += freeMagnitudeToAdd;
+        info.nextPendingFreeMagnitudeIndex += completed;
```

Comment:
This is 2 separate SSTORES. Can we make this a memory variable and store once at the end of the method?

Reply:
(Also it's like 3 SLOADS, kinda defeats the purpose of storage packing)

Reply:
We first load the struct into memory, mutate it then finally mutate storage -> info at the end. 

It appears to be a single sload and sstore unless I'm missing something?

Reply:
when i commented we weren't doing this :D you're right, it's fine now